src/Npgsql/NpgsqlTransaction.cs

```diff
@@ -393,6 +421,32 @@ void CheckReady()
                 throw new InvalidOperationException("This NpgsqlTransaction has completed; it is no longer usable.");
         }
 
+        static bool RequiresQuoting(string identifier)
+        {
+            var first = identifier[0];
+            if (!char.IsLower(first) && first != '_')
+                return true;
+
+            for (var i = 1; i < identifier.Length; i++)
+            {
+                var c = identifier[i];
+
+                if (char.IsLower(c) || char.IsDigit(c))
+                    continue;
+
+                switch (c)
+                {
+                case '_':
+                case '$':  // yes it's true
+                    continue;
+                }
```

Comment:
Can you explain? Do you mean you want me to include the switch in the if above? I did it this way because the switch is possibly a bit more efficient, but I don't mind merging either.

Reply:
Merged your suggestion.